public/app/core/components/AccessControl/Permissions.tsx

```diff
@@ -266,6 +272,16 @@ const setPermission = (
 ): Promise<void> =>
   getBackendSrv().post(`/api/access-control/${resource}/${resourceId}/${type}/${typeId}`, { permission });
 
+const addTeamLBACWarnings = (teams: number[], items: ResourcePermission[]) => {
+  return items.map((item) => {
+    if (item.permission !== 'Admin' && (!item.teamId || !teams.includes(item.teamId))) {
+      item.warning = `Warning: permission does not have corresponding LBAC rule configured.
```

Comment:
This is the main change, and would love this text to be understood easily. I found that the text was a bit hard to understand.

```suggestion
      item.warning = `Warning: Detected LBAC rule. Risk of unrestricted data access, users might have full access to the data. Please use team-based permissions rather than editor/viewer roles on datasources.`
```

Something like this?